### PR TITLE
Remove slot coordination implementation

### PR DESCRIPTION
**Pull request by Asitha**
- Remove slot based message delivery model from Andes core
- Remove thrift
- Implement a single node simple message delivery model

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java [hunk 1]
```diff
@@ -111,28 +91,16 @@ public static MessagingEngine getInstance() {
      * storing strategy will be set according to the configurations by calling this.
      *
      * @param messageStore MessageStore
-     * @param subscriptionManager
      * @throws AndesException
      */
-    public void initialise(MessageStore messageStore, MessageExpiryManager messageExpiryManager,
```
Comment (hastef88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: Why remove subscriptionManager here ?

Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: There were no usages.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/PeriodicExpiryMessageDeletionTask.java [hunk 2]
```diff
@@ -71,81 +61,76 @@
...
-        } catch (AndesException e) {
-            log.error("Error running Message Expiration Checker " + e.getMessage(), e);
-        } catch (InterruptedException e) {
-            log.error("Thread interrupted while waiting for message stores to come online", e);
-        } catch (ExecutionException e) {
-            log.error("Error occurred while waiting for message stores to come online", e);
-        } catch (Throwable e) {
-            log.error("Error occurred during the periodic expiry message deletion task", e);
-        }
+        // TODO: C5 revisit expiration without slot logic
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/PeriodicExpiryMessageDeletionTask.java]: For MB4 can we attempt to move expiration logics away from core?

Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/PeriodicExpiryMessageDeletionTask.java]: hmm... Let's discuss :)

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryTask.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    /**
+     * Flush messages of queue to bounded subscriptions. This will get
+     * a copy of active subscribers at the moment and send messages to them
+     *
+     * @return how many messages sent
+     * @throws AndesException
+     */
+    public int sendMessagesToSubscriptions(StorageQueue queue) throws AndesException {
```
Comment (pamod) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryTask.java]: Let's update java docs as necessary. Definition of  "queue" ?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java [hunk 4]
```diff
@@ -103,16 +89,18 @@
 
 
     public MessageHandler(String queueName) {
-        this.queueName = queueName;
+        this.storageQueueName = queueName;
         ThreadFactory namedThreadFactory = new ThreadFactoryBuilder().
```
Comment (hastef88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java]: Can remove this too ?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java [hunk 5]
```diff
@@ -319,13 +254,6 @@ public int clearReadButUndeliveredMessages() {
         lastPurgedTimestamp = System.currentTimeMillis();
         int messageCount = readButUndeliveredMessages.size();
         readButUndeliveredMessages.clear();
-        for (Slot slot : slotsRead.values()) {
-            if (log.isDebugEnabled()) {
-                log.debug("clear tracking of messages for slot = " + slot);
-            }
-            slot.markMessagesOfSlotAsReturned();
```
Comment (hastef88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java]: How do we update the message status to returned now that this is removed ?

Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java]: We'll have to discuss that. At the moment we don't have something similar to slot returned in standalone mode.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryTask.java (hunk 3 above)
Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryTask.java]: Updated the doc comments

Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageDeliveryTask.java]: Java doc updated

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java (hunk 4 above)
Comment (Asitha) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageHandler.java]: Removed

**Pull request merged**